### PR TITLE
Declared "set_full_screen( )" in all supported backends

### PR DESCRIPTION
**Pull request by Axle7XStriker**
`set_full_screen()` is now declared in all the supported backends and has been implemented in the dummy backend. Now, coverage for the `window.py` can also be increased by implementing tests for `set_full_screen` in the core module.

## PR Checklist:

- [X] All new features have been tested
- [X] All new features have been documented
- [X] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/android/toga_android/window.py [hunk 1]
```diff
@@ -48,6 +48,9 @@ def set_size(self, size):
     def set_app(self, app):
         pass
 
+    def set_full_screen(self, is_full_screen):
```
Comment (freakboy3742) [inline src/android/toga_android/window.py]: The concept of "full screen" doesn't really mean anything on mobile or web - apps are always full screen (or, at least, full viewport). We should flag this as an API endpoint that doesn't exist on Android, iOS and django/web - if you check the dummy backend's app.py, you can see some examples of backend-specific annotation.

**Comment by Axle7XStriker** [comment]
Although I don't know much about this as well but according to me _Window_ should be the one containing this API as in the case of multi-window applications each window will have the freedom to go to full screen mode independently. Although I also agree with your point that it's the app that ultimately goes to full screen mode but it would be much simpler and most logical to have this API on _Window_ rather than _App_. In this case we can simply have a check to avoid _App_ going to full screen mode again and again.

**Comment by tritium21** [comment]
> The mapping of which window goes on which screen might be something you want to modify, but it's the app as a whole that is put into "full screen" mode. Thoughts?

I'm slightly confused, and have different responses based on what full screen means in context (which I think is understandable considering how many UI contexts toga supports).  Does full screen just mean "maximized window"?  In which case, its obvious that set full screen should be on the windows api.  This makes sense for most desktop operating systems, but is meaningless on most mobile platforms.

Does full screen mean "hide desktop UI elements, and take over the full display"?  then that is obviously an api on the application itself... when you are talking about desktop apps.  for mobile... it's still a little bit meaningless, since, that's pretty much the default state for mobile apps.

**Comment by freakboy3742** [comment]
@tritium21 "Full screen" in this context means "overtaking the entire display" - as in the way games or a slide deck operates. 

Part of that API will need to be mapping windows to screens - so, for example, a slide presentation tool will have 2 windows (the slides, and the presenter notes); if there is a secondary screen attached (a monitor, or a projector), the slides go to the monitor/secondary display, and the speaker notes go to the laptop/primary display. 

But, crucially - you can't make *one* window full screen, and have the other *not* be full screen. Either the app is in full screen mode, or it isn't. This, for me, is why the API belongs on the App, not the window.

And yes - it's meaningless for mobile apps, as they essentially always operate in what is effectively "full screen" mode.

**Comment by tritium21** [comment]
> But, crucially - you can't make one window full screen, and have the other not be full screen. Either the app is in full screen mode, or it isn't. This, for me, is why the API belongs on the App, not the window.

... It's been a while since I've had to look, but IIRC Microsoft PowerPoint does just this.  It puts the slides in full screen, but keeps the presenter notes in a normal window.

I know on Windows (the OS), there is a concept of "Windowed fullscreen mode" (gamers are well aware of it for the potential performance boost, but that's besides the point).  It can take over an entire single display (or many displays), from within the windowing system without making lower level calls to take over the hardware.  (You know this is happening usually because starting a game doesn't mess with your second monitor or flash the screen to black, and your desktop icons are exactly where you left them before you started the application, not that I'm bitter or anything).  When an application does this, it can still spawn regular windows on secondary (and tertiary, etc.) displays.

I have no idea how that concept maps onto GTK or OSX.  It's also possible that it sufficiently complicates the code of Toga to support that model, so on those basis, I agree that fullscreen is a concept best mapped to the application, not the window.

**Comment by freakboy3742** [comment]
@tritium21 Thanks for that Windows platform detail - that's good to know. 

However, as you say, it doesn't ultimately change the conclusion - it's an API that still makes sense at the application level. Applications know what windows they have, and which "screens" that are available, so "App.go fullscreen()" is a call to that needs to tell window 1 to go to screen 2, and window 2 to screen 1, and window 3 does nothing - which on macOS would mean it can't be seen, and on Windows would mean you might be able to see it on a third monitor if you had one.

**Comment by Axle7XStriker** [comment]
So, finally according to the discussions, we should migrate the `full_screen` property to _App_ API and implement the logic accordingly. Right?

**Comment by freakboy3742** [comment]
@Axle7XStriker Correct

**Comment by Axle7XStriker** [comment]
Does this look fine or do I need to make some other changes as well to complete the migration process?

File: src/android/toga_android/app.py [hunk 2]
```diff
@@ -49,6 +49,10 @@ def create(self):
 
         self.startup()
 
+    @not_required_on('mobile', 'web')
```
Comment (brutusthebee) [inline src/android/toga_android/app.py]: At column 6: (F821) undefined name 'not_required_on'

File: src/django/toga_django/app.py [hunk 3]
```diff
@@ -35,6 +35,10 @@ def create(self):
     def main_loop(self):
         pass
 
+    @not_required_on('mobile', 'web')
```
Comment (brutusthebee) [inline src/django/toga_django/app.py]: At column 6: (F821) undefined name 'not_required_on'

File: src/iOS/toga_iOS/app.py [hunk 4]
```diff
@@ -104,6 +104,10 @@ def create(self):
         """ Calls the startup method on the interface """
         self.interface.startup()
 
+    @not_required_on('mobile', 'web')
```
Comment (brutusthebee) [inline src/iOS/toga_iOS/app.py]: At column 6: (F821) undefined name 'not_required_on'

**Pull request closed without merging**